src/query/block/scalar.go

```diff
@@ -57,23 +57,30 @@ func (b *Scalar) StepIter() (StepIter, error) {
 	bounds := b.meta.Bounds
 	steps := bounds.Steps()
 	return &scalarStepIter{
-		meta: b.meta,
-		step: scalarStep{
-			vals: []float64{b.val},
-		},
+		meta:    b.meta,
+		s:       b.s,
 		numVals: steps,
 		idx:     -1,
 	}, nil
 }
 
+// ScalarFunc determines whether to return a constant or time t in seconds
+type ScalarFunc func(t time.Time) float64
+
 // SeriesIter returns a SeriesIterator
 func (b *Scalar) SeriesIter() (SeriesIter, error) {
 	bounds := b.meta.Bounds
 	steps := bounds.Steps()
 	vals := make([]float64, steps)
 	for i := range vals {
-		vals[i] = b.val
+		t, err := bounds.TimeForIndex(i)
```

Comment:
can be optimized. t := bounds.Start, and on every iteration, t = t.Add(bounds.StepSize)